build: make error when libsqlite3 is not installed

Make error when `libsqlite3` is not installed.
```bash
❯ make
go build -tags " libsqlite3" -o bin/go-ycsb cmd/go-ycsb/*
# command-line-arguments
/disk1/home/pingyu/opt/go-1.18.5/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: cannot find -lsqlite3
collect2: error: ld returned 1 exit status
```

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 FDB_CHECK := $(shell command -v fdbcli 2> /dev/null)
 ROCKSDB_CHECK := $(shell echo "int main() { return 0; }" | gcc -lrocksdb -x c++ -o /dev/null - 2>/dev/null; echo $$?)
-SQLITE_CHECK := $(shell echo "int main() { return 0; }" | gcc -lsqlite3 -o /dev/null - 2>/dev/null; echo $$?)
+SQLITE_CHECK := $(shell echo "int main() { return 0; }" | gcc -lsqlite3 -x c++ -o /dev/null - 2>/dev/null; echo $$?)
 
 TAGS =
 
@@ -8,7 +8,7 @@ ifdef FDB_CHECK
 	TAGS += foundationdb
 endif
 
-ifdef SQLITE_CHECK
+ifeq ($(SQLITE_CHECK), 0)
 	TAGS += libsqlite3
 endif
 

